app/src/main/java/com/lagradost/cloudstream3/ui/account/AccountHelper.kt

```diff
@@ -0,0 +1,359 @@
+
+        binding.lockProfileCheckbox.setOnCheckedChangeListener { _, isChecked ->
+            if (isChecked) {
+                if (canSetPin) {
+                    showPinInputDialog(context, null, true) { pin ->
+                        if (pin == null) {
+                            binding.lockProfileCheckbox.isChecked = false
+                            return@showPinInputDialog
+                        }
+
+                        currentEditAccount = currentEditAccount.copy(lockPin = pin)
+                    }
+                }
+            } else {
+                if (currentEditAccount.lockPin != null) {
+                    // Ask for the current PIN
+                    showPinInputDialog(context, currentEditAccount.lockPin, true) { pin ->
+                        if (pin == null || pin != currentEditAccount.lockPin) {
+                            canSetPin = false
+                            binding.lockProfileCheckbox.isChecked = true
+                        } else {
+                            currentEditAccount = currentEditAccount.copy(lockPin = null)
+                        }
+                    }
+                }
+            }
+        }
+
+        canSetPin = true
+    }
+
+    fun showPinInputDialog(
+        context: Context,
+        currentPin: String?,
+        editAccount: Boolean,
+        forStartup: Boolean = false,
+        errorText: String? = null,
+        callback: (String?) -> Unit
+    ) {
+        fun TextView.visibleWithText(@StringRes textRes: Int) {
+            isVisible = true
+            setText(textRes)
+        }
+
+        fun TextView.visibleWithText(text: String?) {
+            isVisible = true
+            setText(text)
+        }
+
+        val binding = LockPinDialogBinding.inflate(LayoutInflater.from(context))
+
+        val isPinSet = currentPin != null
+        val isNewPin = editAccount && !isPinSet
+        val isEditPin = editAccount && isPinSet
+
+        val titleRes = if (isEditPin) R.string.enter_current_pin else R.string.enter_pin
+
+        var isPinValid = false
+
+        val builder = AlertDialog.Builder(context, R.style.AlertDialogCustom)
+            .setView(binding.root)
+            .setTitle(titleRes)
+            .setNegativeButton(R.string.cancel) { _, _ ->
+                callback.invoke(null)
+            }
+            .setOnCancelListener {
+                callback.invoke(null)
+            }
+            .setOnDismissListener {
+                if (!isPinValid) {
+                    callback.invoke(null)
+                }
+            }
+
+        if (forStartup) {
+            val currentAccount = DataStoreHelper.accounts.firstOrNull {
+                it.keyIndex == DataStoreHelper.selectedKeyIndex
+            }
+
+            builder.setTitle(context.getString(R.string.enter_pin_with_name, currentAccount?.name))
+            builder.setOnDismissListener {
+                if (!isPinValid) {
+                    exitProcess(0)
```

Comment:
Why is this here???

Reply:
So that if it is wrong, it exits, but only if you close it manually IE cancel button or back button, otherwise it isn't close. It just makes the back button or cancel close instantly rather than showing an empty activity string if trying to enter pin from startup immediately. I can change it to a different method if you want?